includes/Core/Authentication/Clients/OAuth_Client.php

```diff
@@ -539,7 +598,8 @@ public function get_authentication_url( $redirect_url = '' ) {
 		$this->user_options->set( self::OPTION_REDIRECT_URL, $redirect_url );
 
 		// Ensure the latest required scopes are requested.
-		$this->get_client()->setScopes( $this->get_required_scopes() );
+		$scopes = array_merge( $this->get_required_scopes(), $this->get_granted_additional_scopes(), $additional_scopes );
```

Comment:
Per earlier suggestion and conversation with Thierry, we may want to remove `$this->get_granted_additional_scopes()` from right here again so that these scopes are not re-requested next time other scopes are requested (assuming they are likely one-time usage). Storing these additional scopes still makes sense so that the server knows they were granted.

Reply:
I like this idea. I've removed them here so that they are not automatically re-requested in subsequent auth URLs. I have a few ideas as to how we could make this better but it's not something we have bandwidth for in this release so this is a good first step.